libs/template/experimental/virtual-scrolling/src/lib/scroll-strategies/autosized-virtual-scroll-strategy.ts

```diff
@@ -0,0 +1,689 @@
+  /** @internal */
+  private until$ = (o$) => o$.pipe(takeUntil(this.detached$));
+
+  /** @internal */
+  constructor(private differs: IterableDiffers, private ngZone: NgZone) {
+    super();
+  }
+
+  /** @internal */
+  ngOnDestroy() {
+    this.detach();
+  }
+
+  /** @internal */
+  attach(
+    viewport: RxVirtualScrollViewport,
+    viewRepeater: RxVirtualViewRepeater<T, U>
+  ): void {
+    this.viewport = viewport;
+    this.viewRepeater = viewRepeater;
+    this.calcRenderedRange();
+    this.maintainVirtualItems();
+    this.positionElements();
+  }
+
+  /** @internal */
+  detach(): void {
+    this.viewport = null;
+    this.viewRepeater = null;
+    this._virtualItems = [];
+    this.detached$.next();
+  }
+
+  scrollToIndex(index: number, behavior?: ScrollBehavior): void {
+    const _index = Math.min(Math.max(index, 0), this.contentLength - 1);
+    let offset = 0;
+    for (let i = 0; i < _index; i++) {
+      offset += this._virtualItems[i].size || this.tombstoneSize;
+    }
+    this.viewport.scrollTo(offset, behavior);
+  }
+
+  /** @internal */
+  private maintainVirtualItems(): void {
+    this.viewRepeater.values$.pipe(this.until$).subscribe((items: any[]) => {
+      const changes = this.getDiffer(items).diff(items);
+      if (changes) {
+        // reset cache on update
+        this._cachedRange = null;
+        changes.forEachOperation(
+          (item, adjustedPreviousIndex, currentIndex) => {
+            if (item.previousIndex == null) {
+              const entry = {
+                size: 0,
+                tombstone: true,
+              };
+              if (currentIndex < this._virtualItems.length) {
+                this._virtualItems.splice(currentIndex, 0, entry);
+              } else {
+                this._virtualItems.push(entry);
+              }
+            } else if (currentIndex == null) {
+              const removeIdx =
+                adjustedPreviousIndex == null
+                  ? this._virtualItems.length - 1
+                  : adjustedPreviousIndex;
+              removeFromArray(this._virtualItems, removeIdx);
+            } else if (adjustedPreviousIndex !== null) {
+              this._virtualItems[currentIndex] =
+                this._virtualItems[adjustedPreviousIndex];
+            }
+          }
+        );
+      }
+    });
+  }
+
+  /** @internal */
+  private calcRenderedRange(): void {
+    const dataLengthChanged$ = this.viewRepeater.values$.pipe(
+      map(
+        (values) =>
+          (Array.isArray(values)
+            ? values
+            : values != null
+            ? Array.from(values)
+            : []
+          ).length
+      ),
+      distinctUntilChanged()
+    );
+    const onScroll$ = this.viewport.elementScrolled$.pipe(
+      map(() => this.viewport.getScrollTop()),
+      distinctUntilChanged(),
+      startWith(0),
+      tap((_scrollTop) => {
+        this.scrollTop = _scrollTop;
+      })
+    );
+    merge(
```

Comment:
It looks you're only passing one Observable into `merge`, so I guess it can be omitted.